File: Atividade 3/memoria1.js
const fs = require('fs');
const os = require('os');
const { performance } = require('perf_hooks');

function bubbleSort(arr) {
    const n = arr.length;
    for (let i = 0; i < n; i++) {
        for (let j = 0; j < n - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                [arr[j], arr[j + 1]] = [arr[j + 1], arr[j]];
            }
        }
    }
}

function main() {
    const inputFile = "C:\\Users\\rodri\\OneDrive\\Área de Trabalho\\arq.txt";
    const outputFile = "C:\\Users\\rodri\\OneDrive\\Área de Trabalho\\arq-saida.txt";

    // Leitura do arquivo com o BOM ignorado
    const fileContent = fs.readFileSync(inputFile, 'utf8');  // Lê o arquivo como UTF-8
    const numbers = fileContent.split('\n').filter(Boolean).map(Number); // Filtra linhas vazias e converte para números

    // Informações sobre a linguagem e o sistema
    console.log("Linguagem: JavaScript");
    console.log("Versão: " + process.version);
    console.log("Sistema Operacional: " + os.type());
    console.log("Versão do SO: " + os.release());
    console.log("Processador: " + os.cpus()[0].model);
    console.log("Núcleos de CPU: " + os.cpus().length);

    // Uso de memória antes da execução
    const memBefore = process.memoryUsage().heapUsed / 1024; // Em KB
    console.log(`Memória inicial usada: ${memBefore.toFixed(2)} KB`);

    // Ordenação e medição de tempo
    const startTime = performance.now();
    bubbleSort(numbers);
    const elapsedTime = performance.now() - startTime; // Tempo em ms

    // Uso de memória após execução
    const memAfter = process.memoryUsage().heapUsed / 1024; // Em KB

    // Escrita do arquivo de saída no formato correto (um número por linha)
    fs.writeFileSync(outputFile, numbers.map(num => num + '\n').join(''));

    // Exibição dos resultados
    console.log(`Tempo de execução: ${elapsedTime.toFixed(2)} ms`);
    console.log(`Memória usada após execução: ${(memAfter - memBefore).toFixed(2)} KB`);
    console.log(`Arquivo de saída: ${outputFile}`);
}

main();

File: Atividade 3/memoria1.py
import os
import platform
import psutil
import time

def bubble_sort(arr):
    n = len(arr)
    for i in range(n):
        for j in range(0, n - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]

def main():
    input_file = r"C:\Users\rodri\OneDrive\Área de Trabalho\arq.txt"
    output_file = r"C:\Users\rodri\OneDrive\Área de Trabalho\arq-saida.txt"


    with open(input_file, "r", encoding="utf-8-sig") as file:
        numbers = [int(line.strip()) for line in file]


    print(f"Linguagem: Python")
    print(f"Versão: {platform.python_version()}")
    print(f"Sistema Operacional: {platform.system()}")
    print(f"Versão do SO: {platform.release()}")
    print(f"Processador: {platform.processor()}")

    process = psutil.Process(os.getpid())
    mem_before = process.memory_info().rss / 1024  # Memória em KB
    cpu_cores = psutil.cpu_count(logical=True)

    print(f"Núcleos de CPU disponíveis: {cpu_cores}")
    print(f"Memória inicial usada: {mem_before:.2f} KB")

    start_time = time.time()
    bubble_sort(numbers)
    elapsed_time = (time.time() - start_time) * 1000  # Tempo em ms


    mem_after = process.memory_info().rss / 1024  # Memória em KB

    with open(output_file, "w", encoding="utf-8-sig") as file:
        for number in numbers:
            file.write(f"{number}\n")

    print(f"Tempo de execução: {elapsed_time:.2f} ms")
    print(f"Memória usada após execução: {mem_after - mem_before:.2f} KB")
    print(f"Arquivo de saída: {output_file}")

if __name__ == "__main__":
    main()

